app/src/main/java/org/simple/clinic/registration/phone/PhoneNumberValidator.kt

```diff
@@ -28,44 +28,35 @@ class IndianPhoneNumberValidator : PhoneNumberValidator {
   override fun validate(number: String, type: PhoneNumberValidator.Type): Result {
     return when (type) {
       MOBILE -> {
-        when {
-          number.isBlank() -> Blank
-          number.length < 10 -> LengthTooShort(10)
-          number.length > 10 -> LengthTooLong(10)
-          else -> ValidNumber
-        }
+        validateMobile(number, 10, 10, 10)
       }
       LANDLINE_OR_MOBILE -> {
-        when {
-          number.isBlank() -> Blank
-          number.length < 6 -> LengthTooShort(6)
-          number.length > 12 -> LengthTooLong(12)
-          else -> ValidNumber
-        }
+        validateLandlineOrMobile(number, 6, 12)
       }
     }
   }
+
+  fun validateLandlineOrMobile(number: String, startLimit: Int, endLimit: Int): Result {
+    return when {
+      number.isBlank() -> Blank
+      number.length < startLimit -> LengthTooShort(startLimit)
+      number.length > endLimit -> LengthTooLong(endLimit)
+      else -> ValidNumber
+    }
+  }
+
+  fun validateMobile(number: String, limit: Int, smallerThanNumber: Int, greaterThanNumber: Int): Result {
```

Comment:
Is the `limit` parameter necessary here? We can use `smallerThanNumber` and `greaterThanNumber` right? They are the same thing, but we are adding an additional parameter that adds no value.